src/overmap.cpp

```diff
@@ -915,15 +915,26 @@ const oter_id overmap::get_ter(const int x, const int y, const int z) const
     return layer[z + OVERMAP_DEPTH].terrain[x][y];
 }
 
-bool &overmap::seen(int x, int y, int z)
+omt_knowledge_level &overmap::seen(int x, int y, int z)
 {
     if (x < 0 || x >= OMAPX || y < 0 || y >= OMAPY || z < -OVERMAP_DEPTH || z > OVERMAP_HEIGHT) {
-        nullbool = false;
-        return nullbool;
+        nullokl = OKL_UNKNOWN;
+        return nullokl;
     }
     return layer[z + OVERMAP_DEPTH].visible[x][y];
 }
 
+std::string &overmap::terrain_closeup_name(int x, int y, int z)
+{
+    debugmsg("[om::terrain_closeup_name %d %d %d %s]",x,y,z,layer[z + OVERMAP_DEPTH].terrain_closeup_name[point(x,y)].c_str());
+    if (x < 0 || x >= OMAPX || y < 0 || y >= OMAPY || z < -OVERMAP_DEPTH || z > OVERMAP_HEIGHT ||
+        !layer[z + OVERMAP_DEPTH].terrain_closeup_name.count(point(x,y))) {
+        nullstring = "";
+        return nullstring;
+    }
+    return layer[z + OVERMAP_DEPTH].terrain_closeup_name[point(x,y)];
+}
+
```

Comment:
this debugmsg makes using the map tedious. but it also makes the function work. comment out the debugmsg, code stops working.